Dockerfile

```diff
@@ -1,12 +1,19 @@
+#Changed the skeleton order slightly
+
 # Specify base image.
 FROM python:3.9
 
 # Set working directory.
+WORKDIR /code
 
 # Expose port 8000 for your FastAPI server.
-
-# Install Python requirements using Pip.
+EXPOSE 8000
 
 # Copy your codebase into the Docker container.
+COPY /main.py /utils.py /requirements.txt /code/
+
+# Install Python requirements using Pip.
+RUN pip install -r requirements.txt
 
 # Run FastAPI server on the port exposed above.
+CMD uvicorn main:app 
```

Comment:
You need to specify the host address here ([reason](https://stackoverflow.com/a/20778887)).

Also, you should prefer the array-based syntax for CMD over the string-based syntax ([more info here](https://stackoverflow.com/a/20778887)).

Reply:
I have made the said changes. 
I should have cleared this earlier but,
if I run the docker image normally
`docker run myimage`
the application starts, however I can't access it, it says port not exposed 

if I run the docker image like this
`docker run -p 8000:8000 myimage`

the application starts fine and I can access it

but if I create an image without the `EXPOSE 8000` line in dockerfile and run with the -p label, the application still starts and is accessible

Am I missing something?

Reply:
The reason you need to specify `-p` is because you need to publish ports of your container when running it from a Docker image, in order to have them be accessible from the outside world. 
Take a look at this link for more info: https://docs.docker.com/config/containers/container-networking/#published-ports

As for why it still works without the `EXPOSE 8000` line, the `EXPOSE` command doesn't actually do anything other than serving as documentation between the person writing the Dockerfile, and the person running a container from that Dockerfile. If you see `EXPOSE <port_no>` in a Dockerfile, it indicates that you should include `-p <port_num>:<port_num>` when running a container from an image from that Dockerfile.